### ISSUE TITLE
6.9 fresh install - create new Document - JSON-Error

### ISSUE THREAD

**Issue by mediafant** [pimcore/pimcore#8681]
# Bug Report

### Expected behavior
New  created Document will work

### Actual behavior
New Document isn't visible and also parent Document "home" is gone after reload.

### Steps to reproduce
Create new Pimcore
COMPOSER_MEMORY_LIMIT=-1 composer create-project pimcore/skeleton .
./vendor/bin/pimcore-install
Login pimcore-backend and create a new document

Console-Output:
```
Error: You're trying to decode an invalid JSON String:
(...)
{"success":false,"message":"You cannot serialize or unserialize PDO instances","trace":[{"function":"__sleep","class":"PDO","type":"-\u003E","args":[]},{"file":"\/var\/www\/html\/vendor\/symfony\/symfony\/src\/Symfony\/Component\/HttpKernel\/Profiler\/FileProfilerStorage.php","line":175,"function":"serialize","args":[{"token":"537804","parent":null,"children":
(...)
Stack trace:
#0 /var/www/html/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DataCollector/EventDataCollector.php(88): Symfony\Component\VarDumper\Cloner\Data-&gt;offsetSet('called_listener...', Array)
#1 /var/www/html/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DataCollector/EventDataCollector.php(68): Symfony\Component\HttpKernel\DataCollector\EventDataCollector-&gt;setCalledListeners(Array)
#2 /var/www/html/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Profiler/Profiler.php(100): Symfony\Component\HttpKernel\DataCollector\EventDataCollector-&gt;lateCollect()
#3 /var/www/html/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php(112): Symfony\Component\HttpKernel\Profiler\Profiler-&gt;saveProfile(Object(Symfony\Component\HttpKernel\Profiler\Profile))
#4 /var/www/html/ in <b>/var/www/html/vendor/symfony/symfony/src/Symfony/Component/VarDumper/Cloner/Data.php</b> on line <b>165</b><br />
```

Andreas

**Comment by brusch** [comment]
Sorry, I've exactly followed your steps, but I'm not able to reproduce the issue. 
If the issue persists on your side, it would be nice if you could debug this on your and and create a PR for us if it's a general issue. 
Thanks in advance!

**Comment by fracsi** [comment]
@brusch The error can be reproduced on the demo site as well. Just enable debug mode for pimcore.
An example call (only breaks if debug mode is on): https://demo.pimcore.fun/admin/document/tree-get-childs-by-id?node=1

**Comment by brusch** [comment]
I see the problem on the demo instance, but I'm still not able to reproduce it locally. 
Is there anything else that needs to be done, other than enabling debug mode?

**Comment by fracsi** [comment]
For me the error came when i upgraded an existing project from 6.8.12 to 6.9.0.

**Comment by mediafant** [comment]
Hello @brusch - yes I installed pimcore 3 times with this Docker-Compose:

```
version: '2.0'
services:
  redis:
    image: redis:alpine
    container_name: pimcore-redis

  db:
    image: mariadb:10.4
    container_name: pimcore-mariadb
    working_dir: /application
    command: [mysqld, --character-set-server=utf8mb4, --collation-server=utf8mb4_unicode_ci, --innodb-file-format=Barracuda, --innodb-large-prefix=1, --innodb-file-per-table=1]
    volumes:
      - ./db/:/var/lib/mysql
      - ./init/:/docker-entrypoint-initdb.d
    environment:
      - MYSQL_ROOT_PASSWORD=pimcore
      - MYSQL_DATABASE=pimcore
      - MYSQL_USER=pimcore
      - MYSQL_PASSWORD=pimcore

  adminer:
    image: adminer
    volumes:
      - ./init/uploads.ini:/usr/local/etc/php/conf.d/uploads.ini
    ports:
      - 2002:8080

  php:
    image: pimcore/pimcore:PHP7.4-apache
    container_name: pimcore-php
    volumes:
      - ./httpdocs/:/var/www/html:cached
      - ./init/uploads.ini:/usr/local/etc/php/conf.d/uploads.ini
    ports:
     - "2000:80"
     - "2001:443"
    links:
     - db:db
     - redis:redis

  php-debug:
    image: pimcore/pimcore:PHP7.4-apache-debug
    container_name: pimcore-debug-php
    volumes:
      - ./httpdocs/:/var/www/html:cached
      - .init//uploads.ini:/usr/local/etc/php/conf.d/uploads.ini
    ports:
     - "2006:80"
    links:
     - db:db
     - redis:redis
    environment:
      - PHP_IDE_CONFIG="serverName=localhost"
```

**Comment by brusch** [comment]
Maybe someone with the problem and a debugable environment can check this, I'm still not able to get this error on my environment.

**Comment by brusch** [comment]
Seems to be related with https://github.com/pimcore/pimcore/blob/2df5397b68e0ee4a670ced9a6ec72abc163904a2/lib/Model/Listing/Dao/QueryBuilderHelperTrait.php#L23

**Comment by fracsi** [comment]
The error is coming from `\Symfony\Bridge\Doctrine\DataCollector\DoctrineDataCollector`
I have narrowed it down to a case where this DataCollector contains a query which `sql` holds a `QueryBuilder` instance and not a sql statement string.
The QueryBuilder's inner statement is: `SELECT id, type FROM documents WHERE  (parentId = ?) AND published = 1 ORDER BY `index` ASC`.

**Comment by fashxp** [comment]
can you please check, if https://github.com/pimcore/pimcore/pull/8696 fixes the issue for you?

**Comment by fracsi** [comment]
@fashxp It looks like the problem is fixed.
Maybe the  [`ConnectionInterface`](https://github.com/pimcore/pimcore/blob/6.9/lib/Db/ConnectionInterface.php) should add `string` type to the `$query` and `$sql` function params already typehinted as string in the PHPDoc.

**Comment by fashxp** [comment]
adding string type to method signatures is not possible, as methods are implemented directly by dbal connection (and it does not have string type in method signatures).

**Comment by brusch** [comment]
Fixed by #8696
